### PR TITLE
OCPEDGE-812: feat: add tests for cpu limits on workload partitioning

### PR DESCRIPTION
**Pull request by eggfoobar**
workload partitioning now supports cpu limits in resources added a new test and updated existing ones to validate it.

added test to check annotations and resources have been altered updated existing crio check to also validate cpu limits via quota settings

### CODE REVIEW

File: test/extended/cpu_partitioning/crio.go [hunk 1]
```diff
@@ -120,6 +126,10 @@ var _ = g.Describe("[sig-node][apigroup:config.openshift.io] CPU Partitioning no
 		isClusterCPUPartitioned = getCpuPartitionedStatus(oc) == ocpv1.CPUPartitioningAllNodes
 	})
 
+	g.AfterEach(func() {
+		o.Expect(cleanup(managedNamespace, managedNamespace.Namespace())).NotTo(o.HaveOccurred())
```
Comment (ffromani) [inline test/extended/cpu_partitioning/crio.go:130]: nit:
```
o.Expect(cleanup(managedNamespace, managedNamespace.Namespace())).tTo(o.Succeed())
```
Avoiding negatives when possible with inline expectations, a tad easier to process

File: test/extended/cpu_partitioning/managed.go [hunk 2]
```diff
@@ -132,6 +132,67 @@ var _ = g.Describe("[sig-node][apigroup:config.openshift.io] CPU Partitioning cl
...
 	})
+
+	g.Context("with limits", func() {
+
+		var (
+			managedOC = exutil.NewCLI("cpu-partitioning").SetManagedNamespace().AsAdmin()
+		)
+
+		g.AfterEach(func() {
+			o.Expect(cleanup(managedOC, managedOC.Namespace())).NotTo(o.HaveOccurred())
```
Comment (ffromani) [inline test/extended/cpu_partitioning/managed.go:143]: ditto about using positive for inline expectations

File: test/extended/cpu_partitioning/utils.go [hunk 3]
```diff
@@ -47,3 +63,96 @@ func adjustMatcherAndMessageForCluster(isClusterCPUPartitioned bool, matcher t.G
...
+	}
+	return quota
+}
+
+func createResource(cpu, memory string) corev1.ResourceList {
+	return corev1.ResourceList{
+		corev1.ResourceCPU:    resource.MustParse(cpu),
+		corev1.ResourceMemory: resource.MustParse(memory),
+	}
+}
```
Comment (ffromani) [inline test/extended/cpu_partitioning/utils.go:93]: not a fan of such trivial helpers

**Review by ffromani — COMMENTED** [review]
overall looks reasonnable

File: test/extended/cpu_partitioning/crio.go (hunk 1 above)
Comment (eggfoobar) [reply test/extended/cpu_partitioning/crio.go]: Oh fair point, it is easier to read, thanks! Updating

File: test/extended/cpu_partitioning/utils.go (hunk 3 above)
Comment (eggfoobar) [reply test/extended/cpu_partitioning/utils.go]: Oh no problem, removing it.

**Pull request merged**